src/Subject.ts

```diff
@@ -125,8 +125,9 @@ export class Subject<T> extends Observable<T> implements ISubscription {
  * @class AnonymousSubject<T>
  */
 export class AnonymousSubject<T> extends Subject<T> {
-  constructor(protected destination?: Observer<T>, protected source?: Observable<T>) {
+  constructor(protected destination?: Observer<T>, source?: Observable<T>) {
     super();
+    this.source = source;
```

Comment:
@kwonoj 

The previous code will causes this error
```
dist/cjs/src/Subject.ts(127,14): error TS2415: Class 'AnonymousSubject<T>' incorrectly extends base class 'Subject<T>'.
  Property 'source' is optional in type 'AnonymousSubject<T>' but required in type 'Subject<T>'.
```

This is still protected because this `source` is defined in `Observable<T>` and it is inherited. The previous code just only overrides its definitions.

If you don't like this mixture of assignment property, I'll stop to use parameter property for `AnonymousSubject`